make build-integration-test-image fails in cluster/images/etcd.

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


#### What happened:
Building the etcd integration-test-image fails with the following err:
Failure Log:
```
...
Successfully built 36c5619814a0
Successfully tagged k8s.gcr.io/etcd-amd64:3.4.13-1
cp -r /tmp/tmp.qMWcPYXO68 /tmp/tmp.qMWcPYXO68_integration_test
cp Dockerfile /tmp/tmp.qMWcPYXO68_integration_test/Dockerfile
cd /tmp/tmp.qMWcPYXO68_integration_test && sed -i.bak 's|BASEIMAGE|golang:1.12.17|g' Dockerfile
docker build --pull -t etcd-integration-test /tmp/tmp.qMWcPYXO68_integration_test
Sending build context to Docker daemon  250.2MB
Step 1/10 : FROM golang:1.12.17 as builder
1.12.17: Pulling from library/golang
Digest: sha256:851bb81dad374c1abae45bc89e157fae4d5ebbf0f76f5dce16a81dbac0a26c6b
Status: Image is up to date for golang:1.12.17
 ---> ffcaee6f7d8b
Step 2/10 : RUN apt-get update -y       && apt-get -yy -q install --no-install-recommends --no-install-suggests --fix-missing           bash-static
 ---> Using cache
 ---> 23406a1a44ac
Step 3/10 : RUN cp /bin/bash-static /sh
 ---> Using cache
 ---> 3ec8e9d25343
Step 4/10 : FROM RUNNERIMAGE
invalid reference format: repository name must be lowercase
make: *** [Makefile:185: build-integration-test-image] Error 1
```

#### What you expected to happen:
Expected the image build to pass.
#### How to reproduce it (as minimally and precisely as possible):
1. Sync k/k
2. cd cluster/images/etcd
3. make build-integration-test-image
#### Anything else we need to know?:
Change that probably broke it: https://github.com/kubernetes/kubernetes/pull/91171
#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/cluster/images/etcd/Dockerfile
+++ b/cluster/images/etcd/Dockerfile
@@ -12,7 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM BASEIMAGE as builder
+ARG BASEIMAGE
+ARG RUNNERIMAGE
+
+FROM ${BASEIMAGE} as builder
 
 # This image needs bash for running "migrate-if-needed.sh". Instead of a full debian image
 # we use just the bash-static and we wrap bash-static into a distroless image instead of
@@ -23,7 +26,7 @@ RUN apt-get update -y \
 
 RUN cp /bin/bash-static /sh
 
-FROM RUNNERIMAGE
+FROM ${RUNNERIMAGE}
 WORKDIR /
 
 COPY --from=builder /sh /bin/

--- a/cluster/images/etcd/Makefile
+++ b/cluster/images/etcd/Makefile
@@ -143,12 +143,13 @@ else
 	cd $(TEMP_DIR) && echo "ENV ETCD_UNSUPPORTED_ARCH=$(ARCH)" >> Dockerfile
 endif
 
-	# Replace BASEIMAGE with the real base image
-	cd $(TEMP_DIR) && sed -i.bak 's|BASEIMAGE|$(BASEIMAGE)|g' Dockerfile
-	cd $(TEMP_DIR) && sed -i.bak 's|RUNNERIMAGE|$(RUNNERIMAGE)|g' Dockerfile
-
 	# And build the image
-	docker build --pull -t $(REGISTRY)/etcd-$(ARCH):$(IMAGE_TAG) $(TEMP_DIR)
+	docker build \
+		--pull \
+		-t $(REGISTRY)/etcd-$(ARCH):$(IMAGE_TAG) \
+		--build-arg BASEIMAGE=$(BASEIMAGE) \
+		--build-arg RUNNERIMAGE=$(RUNNERIMAGE) \
+		$(TEMP_DIR)
 
 push: build
 	docker tag $(REGISTRY)/etcd-$(ARCH):$(IMAGE_TAG) $(MANIFEST_IMAGE)-$(ARCH):$(IMAGE_TAG)
@@ -181,8 +182,12 @@ unit-test:
 build-integration-test-image: build
 	cp -r $(TEMP_DIR) $(TEMP_DIR)_integration_test
 	cp Dockerfile $(TEMP_DIR)_integration_test/Dockerfile
-	cd $(TEMP_DIR)_integration_test && sed -i.bak 's|BASEIMAGE|golang:$(GOLANG_VERSION)|g' Dockerfile
-	docker build --pull -t etcd-integration-test $(TEMP_DIR)_integration_test
+	docker build \
+		--pull \
+		-t etcd-integration-test \
+		--build-arg BASEIMAGE=golang:$(GOLANG_VERSION) \
+		--build-arg RUNNERIMAGE=$(RUNNERIMAGE) \
+		$(TEMP_DIR)_integration_test
 
 integration-test:
 	docker run --interactive -v $(shell pwd)/../../../:/go/src/k8s.io/kubernetes$(DOCKER_VOL_OPTS) -e GOARCH=$(ARCH) etcd-integration-test \

